rs/tests/testing_verification/testnets/from_config.rs

```diff
@@ -52,34 +107,111 @@ pub fn setup(env: TestEnv) {
     );
 
     if let Some(boundary_nodes) = parsed.boundary_nodes {
-        for bn in boundary_nodes {
-            BoundaryNode::new(bn.name)
-                .allocate_vm(&env)
-                .expect("Allocation of BoundaryNode failed.")
-                .for_ic(&env, "")
-                .use_real_certs_and_dns()
-                .start(&env)
-                .expect("Failed to setup BoundaryNode VM");
-        }
+        boundary_nodes.iter().for_each(|bn| {
+            match bn {
+                ConfigurableBoundaryNode::Simple(bn) => BoundaryNode::new(bn.name.clone()),
+                ConfigurableBoundaryNode::Complex(b) => *b.to_owned(),
+            }
+            .allocate_vm(&env)
+            .expect("Allocation of BoundaryNode failed.")
+            .for_ic(&env, "")
+            .use_real_certs_and_dns()
+            .start(&env)
+            .expect("Failed to setup BoundaryNode VM")
+        })
     }
 
     env.sync_with_prometheus();
 }
 
+fn write_file_and_update_env_variable(env: &TestEnv, pairs: Vec<(&str, String, &str)>) {
+    for (file_name, value_in_file, env_variable) in pairs {
+        let path = env.get_dependency_path(file_name);
+        std::fs::write(&path, value_in_file)
+            .unwrap_or_else(|_| panic!("Failed to write to path: {}", path.display()));
+        std::env::set_var(env_variable, file_name);
+        info!(
+            env.logger(),
+            "Overriden env variable `{}` to value: {}",
+            env_variable,
+            path.display()
+        )
+    }
+}
+
+fn fetch_shasum_for_disk_img(version: String) -> String {
+    let url = format!(
+        "http://download.proxy-global.dfinity.network:8080/ic/{}/guest-os/disk-img/SHA256SUMS",
+        version
+    );
+    let response = reqwest::blocking::get(&url)
+        .unwrap_or_else(|e| panic!("Failed to fetch url `{}` with err: {:?}", &url, e));
+    if !response.status().is_success() {
+        panic!(
+            "Received non-success response status: {:?}",
+            response.status()
+        )
+    }
+
+    String::from_utf8(
+        response
+            .bytes()
+            .expect("Failed to deserialize bytes")
+            .to_vec(),
+    )
+    .expect("Failed to convert to UTF8")
+    .lines()
+    .into_iter()
+    .find(|l| l.ends_with(TAR_EXTENSION))
+    .unwrap_or_else(|| {
+        panic!(
+            "Failed to find a hash ending with `{}` from: {}",
+            &url, TAR_EXTENSION
+        )
+    })
+    .split_whitespace()
+    .into_iter()
+    .next()
+    .expect("The format of hash should contain whitespace")
+    .to_string()
+}
+
```

Comment:
Managed to do it how you suggested. The good thing is that fetching `sha` will fail if the version is not present so we cover that path as well 💯